docs/python_support/python_support.md

```diff
@@ -0,0 +1,333 @@
+def execute(self, inputs):
+    # For single set on inputs generate 10 sets of outputs
+    for _ in range(10):
+        ... 
+        yield outputs
+```
+
+More information along with the configuration aspect described can be found in [execution modes]() section.
+
+#### Parameters and return value
+
+`execute` is called with `inputs` parameter which is a `list of pyovms.Tensor`. 
+
+Depending on the mode it should return:
+
+- For regular mode: `list of pyovms.Tensor`
+- For generative mode: `generator` that generates `list of pyovms.Tensor`
+
+So depending on the mode `execute` must always either `return` or `yield` a `list of pyovms.Tensor`
+
+*Note*: This method returns outputs as a full set, but since each output is a separate packet in Mediapipe flow, they do not arrive together to their destination. Be aware that if you have more than one output and outputs of your Python node are also outputs of the whole graph you will receive each output in a separate response. You can then gather them using timestamp that can be found in the response. TODO: more about it...
+
+#### Error handling
+
+Signaling that something went wrong should be done by throwing an exception.
+The exception is caught by the `PythonExecutorCalculator` which logs it and returns non-OK status.
+Model Server then reads that status and sets graph in an error state. Then it closes all graph's input streams and waits until in-progress actions are finished. Once it's done the graph gets removed.
+
+This behavior has different effect on the client depending on the kind of gRPC endpoint used - unary or streaming.
+
+**Unary** 
+
+With unary endpoint a graph is created, executed and destroyed for every request. If `execute` encounters an error, model server logs it and sends error message in response immediately. 
+
+**Streaming**
+
+With streaming endpoint a graph is created for the first request in the stream and then reused by all subsequent requests. 
+
+If `execute` encounters an error on the first request (for example the Python code doesn't work as expected), model server logs it  and sends error message in response immediately. The graph gets destroyed.
+
+If `execute` encounters an error on one of the subsequent requests (for example wrong data has been received), model server logs it and Mediapipe sets error in the graph, but the client won't receive error message until it sends another request. When the next request is read from the stream, model server checks if graph has an error, destroys it and sends response to the client.
+
+As of now, the graphs are not recoverable so if an error occurs, you need to create a new stream.
+
+**Implementing this function is required.**
+
+### finalize
+
+`finalize` is called when model server unloads graph definition. It allows to perform any cleanup actions before the graph is removed. 
+
+#### Parameters and return value
+
+`finalize` does not have any parameters and is not expected to return any value.
+
+#### Error handling
```

Comment:
Subjective, but I found it misleading why so many error handling paragraphs are here. It might be helpful to distinguish those init/exec/final paragraphs even more